### ISSUE TITLE
In COMPASS, enable setup_testcase.py to clone a git repository

### ISSUE THREAD

**Issue by sbrus89** [MPAS-Dev/MPAS-Model#383]
This is just a minor feature that I thought might be useful and was interested to see if anyone else thinks it's worth pursuing at some point.

For example, I use scripts from MPAS-QuickViz to do analysis for a testcase. It would be nice if I could make the step to clone this repo a part of running setup_testcase.

**Comment by pwolfram** [comment]
What about cloning the repo as part of an analysis step for a case, where you check to see if the repo exist with the right checksum, and if not, then you clone it?  I don't think this should go in setup_testcase.py.

**Comment by xylar** [comment]
We just talked on the phone but here's the written version for posterity.

If you need something from MPAS-QuickViz in COMPASS, it should be moved to MPAS-Tools instead.   Basically, something like this seems like a step back toward having paths to scripts.  Anything like that should either be in COMPASS or be part of one of its conda dependencies in my view.   Versioning can also potentially be tricky -- you'd have to point to a specific hash or release rather than a branch for the cloning approach to be safe.   Another issue could be if the test case is being set up when there's no internet access.

Long story short, I don't like the idea of cloning a repo as part of `setup_testcase.py`.

**Comment by pwolfram** [comment]
This makes sense-- it would be better to store the script in COMPASS (no version issues) or if it is more stable and less specific to the simulation in MPAS-Tools.  My vote is for it to be directly in COMPASS in a general scripts repo.

**Comment by sbrus89** [comment]
I agree with you both. I don't know why I didn't think of doing it that way, i.e. committing the scripts to COMPASS or MPAS-Tools. Thanks for the feedback. I'll go ahead and close this out!
